### ISSUE TITLE
"Virtual vertices" would make fixing up existing ways much easier

### ISSUE THREAD

**Issue by tomhughes** [tomhughes/trac-tickets#821]
**[Submitted to the original trac issue database at 11.04pm, Sunday, 6th April 2008]**

Copied from http://josm.openstreetmap.de/ticket/595 (exchange JOSM with Merkaartor):

```
19:11:18 crschmidt * RichardF: http://openlayers.org/dev/examples/modify-feature.html
19:11:20 crschmidt * draw a polygon
19:11:29 crschmidt * then click 'modify feature'
19:11:40 crschmidt * you'll see that on each side, there are extra, draggable points
```

I would love to see that in JOSM. I guess especially the AND and TIGER fixup guys would give all sorts of gratitude :)

If that link doesn't work: imagine you have a way like the following

*-----+------*------+------*

where the * are real nodes, it draws the + as selection points that you can drag, and when you release the mouse, it becomes a real new node with two new + on either side of it, ready for the next drag.

**Comment by tomhughes** [comment]
**Author: koying**
**[Added to the original trac issue at 9.43am, Friday, 11th July 2008]**

Demoted to "Minor" due to non-functional request

**Comment by tomhughes** [comment]
**Author: ColinMarquardt**
**[Added to the original trac issue at 9.28pm, Thursday, 11th September 2008]**

For what it's worth, JOSM got this feature a while ago, and users seem to love it.

**Comment by tomhughes** [comment]
**Author: koying**
**[Added to the original trac issue at 7.42pm, Monday, 15th September 2008]**

I read some nice comments about it on the dev list.

While it seems much work just to avoid pressing CTRL-N and clicking, it is worth considering.

**Comment by tomhughes** [comment]
**Author: koying**
**[Added to the original trac issue at 2.36am, Monday, 4th January 2010]**

(In [19269]) ADD : Virtual nodes allow fast creation of intermediary nodes (closes #821)
